src/crypto/x509/x509.go

```diff
@@ -1787,13 +1787,15 @@ func buildCertExtensions(template *Certificate, subjectIsEmpty bool, authorityKe
 	}
 
 	// This extension is not critical
-	if template.KeyUsage == KeyUsageDigitalSignature && template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
-		ret[n].Id = oidExtensionDelegatedCredential
-		ret[n].Value, err = asn1.Marshal(asn1.NullRawValue)
-		if err != nil {
-			return
+	if template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
+		if template.KeyUsage&KeyUsageDigitalSignature != 0 {
```

Comment:
Why next this if statement? I think it's clearer if you move the condition into the if statement above.

Reply:
Solved.